### DISCUSSION TITLE
The next version of Octokit.net

### DISCUSSION THREAD

**Discussion by nickfloyd** [octokit/octokit.net#2824] [General]
_Pulled from this [issue](https://github.com/octokit/octokit.net/issues/2737#issuecomment-1832627395)_

We are in the middle of a seriously exciting effort where we are using the incredible work that our Microsoft friends have done ([Kiota](https://github.com/microsoft/kiota)) and are leaning heavily into generating our SDKs - We https://github.com/octokit/octokit.net/discussions/2499 to this over a year ago and have continued the discussion https://github.com/octokit/octokit.net/discussions/2495 and https://github.com/octokit/octokit.net/discussions/2527).

So what does this mean?

We'll get close to 100% GitHub REST API coverage, have an SDK that supports REST API versioning (lowers the risk of breaking changes), support more Auth patterns, and begin to unlock a host of other things/patterns that we were previously not able to do. Oh and we'll be supporting more languages eventually.

When is this happening?

Hopefully sooner than later. We've been working through some [prototyping](https://github.com/octokit/source-generator) and are fairly close on getting our thoughts down as code. The goal is to get this things as "right" as we can so we are being measured about how we approach this so that it will be beneficial to the community.

Please let @kfcampbell or me (@nickfloyd) know if y'all have any questions or want more information.

❤️ We genuinely need and want your feedback and involvement.  Please consider adding your use cases, needs, thoughts, opinions, and anything else you can think of that will help make this something the community needs and wants

**Comment by martincostello** [comment]
I don't know how well the approach would transfer over to C#, but I'd like to point out that I've found using Octokit.js in TypeScript really nice to use to get scripting done, particularly the way it composes up with the plugins so you can do things like automatically paginate, handling GitHub API's caching to minimise the number of rate limit "units" you use to do something (e.g. when polling for the merge status of a PR), retry etc.

It would be great if whatever of those features make sense for .NET/C# could similarly make its way over into the Picard era of Ocktokit.net.

**Comment by nickfloyd** [comment]
Since this information impacts other SDKs in other languages, I'd like to redirect the conversation here: https://github.com/orgs/octokit/discussions/71
